builtin_array.go

```diff
@@ -794,6 +795,7 @@ func (r *Runtime) arrayproto_map(call FunctionCall) Value {
 					values[k] = callbackFn(fc)
```

Comment:
[important] the corresponding value that is created is tracked within the callbackFn function because the value is pushed on the stack BUT that value is [popped](https://github.com/mongodb-forks/goja/blob/b528e7ba6d7ab093bc639ee5c6dd97afd633bf6f/func.go#L334) off the stack at the end of the function call.

The only difference is now instead of just stopping tracking this value altogether, we are moving it to be tracked in the `tmpValues` array.